src/unix/core.c

```diff
@@ -1648,3 +1649,35 @@ int uv__search_path(const char* prog, char* buf, size_t* buflen) {
   /* Out of tokens (path entries), and no match found */
   return UV_EINVAL;
 }
+
+
+unsigned int uv_available_parallelism(void) {
+#ifdef __linux__
+  cpu_set_t set;
+  long rc;
+
+  memset(&set, 0, sizeof(set));
+
+  /* sysconf(_SC_NPROCESSORS_ONLN) in musl calls sched_getaffinity() but in
+   * glibc it's... complicated... so for consistency try sched_getaffinity()
+   * before falling back to sysconf(_SC_NPROCESSORS_ONLN).
+   */
+  if (0 == sched_getaffinity(0, sizeof(set), &set))
+    rc = CPU_COUNT(&set);
+  else
+    rc = sysconf(_SC_NPROCESSORS_ONLN);
+
+  if (rc < 1)
+    rc = 1;
```

Comment:
Returning 1 feels misleading to me. I would think either returning error or the length of the uv_cpu_info array would be more useful.

Reply:
I respectfully disagree for two reasons, one philosophical, one pragmatic:

1. I want this function to be simple to use. Error handling complicates that and adds no value, it just forces callers to duplicate this logic.

2. `sysconf(_SC_NPROCESSORS_ONLN)` can't really fail, to the best of my knowledge. I've sampled a couple of libcs and they all return 1 as a fallback (except glibc - glibc returns 2 because of course it does.)

Reply:
okay, that is comelling